R/pkg/R/functions.R

```diff
@@ -2638,3 +2638,81 @@ setMethod("sort_array",
             jc <- callJStatic("org.apache.spark.sql.functions", "sort_array", x@jc, asc)
             column(jc)
           })
+
+#' This function computes a histogram for a given SparkR Column.
+#' 
+#' @name histogram
+#' @title Histogram
+#' @param nbins the number of bins (optional). The default is 10.
+#' @param df the DataFrame containing the Column to build the histogram from.
+#' @param colname the name of the column to build the histogram from.
+#' @return a data.frame with the histogram statistics, i.e., counts and centroids.
+#' @examples \dontrun{
+#' 
+#' # Create a DataFrame from the Iris dataset
+#' irisDF <- createDataFrame(sqlContext, iris)
+#' 
+#' # Compute histogram statistics
+#' histData <- histogram(df, "colname"Sepal_Length", nbins = 12)
+#'
+#' # Once SparkR has computed the histogram statistics, it would be very easy to
+#' # render the histogram using R's visualization packages such as ggplot2.
+#'   
+#' } 
+setMethod("histogram",
+          signature(df = "DataFrame"),
+          function(df, colname, nbins = 10) {
```

Comment:
@felixcheung Yeah, I thought of that but I don't know how to compute the min and max in one single pass given a Column (not a name). I used describe() which requires a column name. I also tried agg, but it cannot compute more than 1 stat per column:

```
> collect(agg(irisDF, Sepal_Length="max", Sepal_Width="min"))
  max(Sepal_Length) min(Sepal_Width)
1               7.9                2

> collect(agg(irisDF, Sepal_Length="max", Sepal_Length="min"))
  max(Sepal_Length)
1               7.9

Suggestions?